lib/src/facts/cache.cc

```diff
@@ -0,0 +1,116 @@
+#include <internal/facts/cache.hpp>
+#include <internal/facts/external/json_resolver.hpp>
+
+#include <rapidjson/document.h>
+#include <rapidjson/prettywriter.h>
+#include <leatherman/logging/logging.hpp>
+#include <leatherman/file_util/directory.hpp>
+#include <leatherman/file_util/file.hpp>
+#include <boost/nowide/fstream.hpp>
+#include <time.h>
+
+using namespace std;
+using namespace rapidjson;
+using namespace facter::facts::external;
+namespace boost_file = boost::filesystem;
+
+namespace facter { namespace facts { namespace cache {
+
+    bool cache_is_valid(boost_file::path const& cache_dir, int64_t ttl) {
+        time_t last_mod = boost_file::last_write_time(cache_dir);
```

Comment:
@Magisus do the contents of a given file map directly back to the config file? that is, i would expect that if i configure 

`facts : { ttls : [ { "os" : 30 days } ] }`

all of the `os` facts would be collected in a batch and kept in one file. I *think* that's what you're saying - if so i am enthusiastic 👍